### DISCUSSION TITLE
Support a different view macro

### DISCUSSION THREAD

**Discussion by yerlaser** [leptos-rs/leptos#668] [Ideas]
I understand that the JSX like view macro was chosen due to familiarity and popularity among web developers.

However, I guess you realize that JSX has many shortcomings, let me name a few of them:
1. XML tags are very verbose and often less readable than other template languages, compare `<button>text</button>` vs Sycamore's syntax: `button { "text" }`
2. It looks odd in a Rust code which causes constant mental switches
3. Only a few (mostly rather heavy) editors support language injection. Thus, things like commenting lines, indentation, syntax highlighting aren't available in many good code editors.
4. Again, unless the code editor supports automatic tag pair closure, deletion and renaming, dealing with tag pairs becomes a hassle and can lead to errors when the start and end tags are not in sync.

Given above, do you think Leptos can support a different, e.g. Sycamore style view macro? This can be switched on through a different Cargo.toml entry, different import, macro name or any other way.

**Comment by gbj** [comment]
Yeah I mean... points 1 and 2 are subjective and I disagree, which is why I chose the syntax I did rather than something like Sycamore or Dioxus's custom template DSLs. 

The view macro is converted into a builder. So you could either use the builder syntax directly, or implement a macro to convert from the syntax of your choice into the builder. It's not likely that I'll personally spend any time creating a macro for an alternate syntax/custom DSL.

**Reply by yerlaser** [reply]
OK. Fair enough. Thanks for the response.
